src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -203,6 +207,69 @@ func (r *Cluster) validateKafkaListeners() field.ErrorList {
+	var allErrs field.ErrorList
+	var proxyExternal *PandaproxyAPI
+	kafkaExternal := r.ExternalListener()
+	for i, p := range r.Spec.Configuration.PandaproxyAPI {
+		if !p.External.Enabled {
+			continue
+		}
+		if proxyExternal != nil {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"only one pandaproxy api listener can be marked as external"))
+		}
+		proxyExternal = &r.Spec.Configuration.PandaproxyAPI[i]
+		if proxyExternal.Port != 0 {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"external pandaproxy api listener cannot have port specified, it's autogenerated"))
+		}
+		if (kafkaExternal == nil || !kafkaExternal.External.Enabled) && (proxyExternal != nil && proxyExternal.External.Enabled) {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"cannot have a pandaproxy external listener without a kafka external listener"))
+		}
+		if kafkaExternal == nil && proxyExternal.External.Subdomain != "" {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"kafka external listener must be specified with the same sudomain as that of the external pandaproxy"))
+		}
+		if kafkaExternal != nil && kafkaExternal.External.Subdomain != proxyExternal.External.Subdomain {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"sudomain of external pandaproxy must be the same as kafka's"))
+		}
+	}
+
+	// TLS for pandaproxy is not supported yet
+	for i, p := range r.Spec.Configuration.PandaproxyAPI {
+		if p.TLS.Enabled {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi").Index(i).Child("tls"),
+					r.Spec.Configuration.PandaproxyAPI[i].TLS,
+					"no listener can have TLS enabled"))
+			// we need to run the validation on all listeners to also catch errors like !Enabled && RequireClientAuth
+			allErrs = append(allErrs, validatePandaproxyTLS(p.TLS, field.NewPath("spec").Child("configuration").Child("pandaproxyApi").Index(i).Child("tls"))...)
+		}
+	}
+
+	if !((len(r.Spec.Configuration.PandaproxyAPI) == 2 && proxyExternal != nil) || (proxyExternal == nil && len(r.Spec.Configuration.PandaproxyAPI) <= 1)) {
+		allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+				r.Spec.Configuration.PandaproxyAPI,
+				"up to one internal listener and no external listener, or one external and one internal listener for pandaproxy is allowed"))
```

Comment:
is internal listener "up to one" or exactly one? Because I think in some places in code you expect that internal listener exists (e.g. when computing external port - you expect that internal port exists as well and external is internal+1)

Reply:
"up to one". The callers of `PandaproxyAPIInternal()` check for nil.

These are the valid cases for pandaproxy ports:
```
(internal, external)
0,0
1,0
1,1
```